### DISCUSSION TITLE
Contributing

### DISCUSSION THREAD

**Discussion by renebarto** [rsta2/circle#496] [General]
Hi Rene, we discussed a few things in the past, and I was thinking to add CMake support to the build infrastructure. However, as far as I can see I cannot create a branch. I see there are contributors, so how can I contribute to Circle?
I also found a bug in the rotary switch handling some time ago, but could not fix it for the same reason.

Regards, and thanks for the great work,

Rene Barto

**Comment by rsta2** [comment]
Hi Rene,

thanks for appreciating Circle!

To contribute you have to create a Pull Request (PR) on GitHub. You cannot directly push to the Circle repository. To create a PR you have to create your own fork of the Circle repo in your project space. Then clone it to your local machine, create a new branch, based on the *develop* branch (!), and apply your changes on this branch. Then commit the changes, push the branch to GitHub, login into GitHub and create a PR to be merged into the base branch *develop* of Circle.

Changing the build system is a big step for a project. I think adding CMake support would require a new *CMakeList.txt* file in each directory. This needs not only to be created, it must be maintained over time. Given that I'm fine with the current build system, it's lean and mean, I'm not sure, if I would be happy about this modification. We had this discussion earlier, and I decided against it at this time.

Rene

**Reply by renebarto** [reply]
Hi Rene, clear, I know what to do and understand your concerns. Anyway, I can submit the fix for the rotary controller to start with. My guess was already to create a fork, so I'll simple go along with that.

**Comment by rsta2** [comment]
@renebarto I'm interested in the fix for the rotary encoder. Can you tell me, what the problem is in this driver?
